### PR TITLE
DT: Ensure n_folds and null_proxy are set correctly

### PR DESCRIPTION
**Pull request by iyerr3**
The summary table in Decision Tree included two entries: k and
null_proxy. The 'k' value is supposed to reflect the 'n_folds' value but
was not set correctly and was always NULL. The null_proxy was supposed
to be NULL when null_as_category=False but was set to the string 'NULL'.
Both these issues have been corrected in this commit.

Closes #236

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/346/

File: src/ports/postgres/modules/recursive_partitioning/decision_tree.py_in [hunk 1]
```diff
@@ -1463,15 +1463,15 @@ def _create_summary_table(
         cp_str = py_list_to_sql_string(cp, 'double precision')
     else:
         cp_str = str(cp) + "::double precision"
-    fold = k if running_cv else "NULL"
+    # n_folds_str = n_folds if running_cv else "NULL"
```
Comment (njayaram2) [inline src/ports/postgres/modules/recursive_partitioning/decision_tree.py_in]: Can we remove this commented line?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/352/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/357/

**Pull request closed without merging**